app/src/main/java/com/example/kuba10/mypokemonplaces/ChooseFragment/ChooseFragment.java

```diff
@@ -1,98 +0,0 @@
-package com.example.kuba10.mypokemonplaces.ChooseFragment;
-
-
-import android.content.Context;
-import android.os.Bundle;
-import android.support.v4.app.Fragment;
-import android.support.v7.widget.GridLayoutManager;
-import android.support.v7.widget.RecyclerView;
-import android.view.LayoutInflater;
-import android.view.View;
-import android.view.ViewGroup;
-import android.widget.LinearLayout;
-
-import com.example.kuba10.mypokemonplaces.Constants;
-import com.example.kuba10.mypokemonplaces.FragmentListener;
-import com.example.kuba10.mypokemonplaces.Model.PokemonGo;
-import com.example.kuba10.mypokemonplaces.R;
-
-import java.util.ArrayList;
-
-
-public class ChooseFragment extends Fragment {
-
-
-    private PokemonImageAdapter pokeAdapter;
-    private GridLayoutManager gridManager;
-    private RecyclerView recyclerView;
-    private LinearLayout errorImage;
-    private FragmentListener fragmentListener;
-    private ArrayList<PokemonGo> pokemonGo_data_list;
-
-
-    public static ChooseFragment newInstance(ArrayList<PokemonGo> pokemonGo_data_list) {
-        ChooseFragment fragment = new ChooseFragment();
-
-        Bundle args = new Bundle();
-        args.putParcelableArrayList(Constants.LIST, pokemonGo_data_list);
-        fragment.setArguments(args);
-        return fragment;
-
-
-    }
-
-    @Override
-    public void onAttach(Context context) {
-        super.onAttach(context);
-
-        fragmentListener = (FragmentListener) context;
-        pokemonGo_data_list = getArguments().getParcelableArrayList(Constants.LIST);
-    }
-
-    @Override
-    public void onStart() {
-        super.onStart();
-
-        if (pokemonGo_data_list.size() == 0) {
-            errorImage.setVisibility(View.VISIBLE);
-
-
-        }
-    }
-
-
-    @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState) {
-
-        View view = inflater.inflate(R.layout.fragment_choose, container, false);
-        errorImage = (LinearLayout) view.findViewById(R.id.error_image_gallery);
-
-
-        recyclerView = (RecyclerView) view.findViewById(R.id.choose_recycler_view);
-        gridManager = new GridLayoutManager(getActivity(), 4);
-        recyclerView.setLayoutManager(gridManager);
-
-
-        pokeAdapter = new PokemonImageAdapter(getActivity(), fragmentListener, this);
-        recyclerView.setAdapter(pokeAdapter);
-
-        pokeAdapter.notifyDataSetChanged();
-        errorImage.setVisibility(View.GONE);
-
-        return view;
-
-    }
-
-
-    @Override
-    public void onDetach() {
-        super.onDetach();
-        fragmentListener = null;
-    }
-
-    public void dismiss() {
```

Comment:
Nie za bardzo wiem o co z tym chodzi, tak jak wyżej, ale skoro jest jakaś metoda powtarzana w kilku klasach to robimy do niej abstrakcję - albo klasę nadrzędną albo nadrzędną abstrakcyjną

Reply:
To jest abstrakcja - metoda jest w main activity, a każdy z fragmentów się do niej odwołuje przez interface - chyba że jest jakiś lepszy sposób zamykania fragmentów niż przez fragmentManager z klasy głównej?

Reply:
Ok, wrócimy do tego przy kolejnej iteracji review